similar/editor/medwall.cpp

```diff
@@ -410,14 +410,15 @@ window_event_result wall_dialog_handler(UI_DIALOG *dlg,const d_event &event, wal
 	//------------------------------------------------------------
 	ui_button_any_drawn = 0;
 
-	const auto &&w = vwallptr(Cursegp->sides[Curside].wall_num);
+	const auto wn = Cursegp->sides[Curside].wall_num;
```

Comment:
This looks like a good use case for `wallptridx` (not `vwallptridx`).  You can construct a `wallptridx` from an index that is equal to None or is a valid index; any invalid value other than None triggers an exception.  You can treat a `wallptridx` like a pointer in most cases, and it spares you keeping a separate explicit copy of the index in `wn`, because you can also treat it like an index when needed.